subprojects/packagefiles/sdl2/meson.build

```diff
@@ -10,375 +14,1034 @@ version_micro = varr[2]
+    endif
 endif
 
+core_subsystems = {
+    'atomic'        : { 'opt' : opt_atomic,     'deps' : [], 'opt_deps' : [] },
+    'audio'         : { 'opt' : opt_audio,      'deps' : [], 'opt_deps' : [alsa_dep, libpulse_simple_dep, jack_dep, libpipewire_dep] },
+    'cpuinfo'       : { 'opt' : opt_cpuinfo,    'deps' : [], 'opt_deps' : [] },
+    'events'        : { 'opt' : opt_events,     'deps' : [], 'opt_deps' : x11_deps + wayland_deps },
+    'file'          : { 'opt' : opt_file,       'deps' : [], 'opt_deps' : [] },
+    'filesystem'    : { 'opt' : opt_filesystem, 'deps' : [], 'opt_deps' : [] },
+    'haptic'        : { 'opt' : opt_haptic,     'deps' : [], 'opt_deps' : [] },
+    'hidapi'        : { 'opt' : opt_hidapi,     'deps' : [], 'opt_deps' : [libusb_dep] },
+    'joystick'      : { 'opt' : opt_joystick,   'deps' : [], 'opt_deps' : [] },
+    'loadso'        : { 'opt' : opt_loadso,     'deps' : [], 'opt_deps' : [] },
+    'locale'        : { 'opt' : opt_locale,     'deps' : [], 'opt_deps' : [] },
+    'power'         : { 'opt' : opt_power,      'deps' : [], 'opt_deps' : [] },
+    'render'        : { 'opt' : opt_render,     'deps' : [], 'opt_deps' : [vulkan_dep] + x11_deps + wayland_deps + gl_deps },
+    'sensor'        : { 'opt' : opt_sensor,     'deps' : [], 'opt_deps' : [] },
+    'threads'       : { 'opt' : opt_threads,    'deps' : [threads_dep], 'opt_deps' : [] },
+    'timers'        : { 'opt' : opt_timers,     'deps' : [], 'opt_deps' : [] },
+    'video'         : { 'opt' : opt_video,      'deps' : [], 'opt_deps' : [vulkan_dep] + x11_deps + wayland_deps + gl_deps },
+}
+
+enabled_subsystems = []
+
+foreach ss_name, subsystem : core_subsystems
+    ss_enabled = true
+
+    if not subsystem['opt'].disabled()
+        foreach dep : subsystem['deps']
+            if not dep.found()
+                warning('Subsystem "@0@" disabled due to missing dependency "@1@"'.format(ss_name, dep.name()))
+
+                if subsystem['opt'].enabled()
+                    error('Subsystem @0@ couldn\'t be enabled'.format(ss_name))
+                endif
+
+                ss_enabled = false
+            endif
+        endforeach
+    else
+        ss_enabled = false
+    endif
+
+    if ss_enabled
+        foreach dep : subsystem['deps']
+            if dep.found() and not extra_deps.contains(dep)
+                core_deps += dep
+            endif
+        endforeach
+
+        foreach dep : subsystem['opt_deps']
+            if dep.found() and not extra_deps.contains(dep)
+                extra_deps += dep
+            endif
+        endforeach
+
+        message('Subsystem "@0@" is ENABLED'.format(ss_name))
+        enabled_subsystems += ss_name
+    else
+        cdata.set('SDL_@0@_DISABLED'.format(ss_name.to_upper()), 1)
+        message('Subsystem "@0@" is DISABLED'.format(ss_name))
+    endif
+endforeach
+
+# sigh...
+# https://github.com/mesonbuild/meson/issues/1889
+# Let's try to cure the superproject std
+unfuck_std = '''
```

Comment:
This variable name should be something different.

Reply:
removed the obscene language and updated the comment to be more descriptive on what its doing